src/presentation/http/router/note.ts

```diff
@@ -111,14 +111,26 @@ const NoteRouter: FastifyPluginCallback<NoteRouterOptions> = (fastify, opts, don
     ],
   }, async (request, reply) => {
     const { note } = request;
+    const noteId = request.note?.id as number;
     const { memberRole } = request;
+    const { userId } = request;
 
     /**
      * Check if note exists
      */
     if (note === null) {
       return reply.notFound('Note not found');
     }
+
+    /**
+     * Check if user is authorized
+     * We can not move this to noteService because this we get note from NoteResolver
+     * NoteResolver has no information about user to save noteVisit
+     */
+    if (userId !== null) {
+      await noteService.saveVisit(noteId, userId);
```

Comment:
why have you left this on a presentation level?

Reply:
there is comment about it right above

Reply:
This is not good, that you have different patterns in different routes